### ISSUE TITLE
Can't connect / timeout (Postgresql in docker container)?

### ISSUE THREAD

**Issue by dmitry-saritasa** [ClearcodeHQ/pytest-postgresql#57]
Python 3.5.2 
Postgresql 9.6.2 running inside docker container with exposed ports (so I can run psql locally)
Port: 5432

# tests are blocked

```
(storm) ➜  src git:(develop) ✗ pytest -v 
============================================================================================================= test session starts =============================================================================================================
platform linux -- Python 3.5.2, pytest-3.0.7, py-1.4.33, pluggy-0.4.0 -- /home/dmitry/.pyenv/versions/3.5.2/envs/storm/bin/python
cachedir: .cache
rootdir: /home/dmitry/Projects/trivver/storm/src, inifile:
plugins: xdist-1.16.0, postgresql-1.2.0, cov-2.4.0
collected 5 items 

cache/tests/lru_test.py::TestCacheLRU::test_get_session_user  ^C
```
So my execution stuck inside test_get_session_user()

**code**:
```
postgresql_proc = factories.postgresql_proc(user='postgres')
postgresql = factories.postgresql('postgresql_proc')


class TestCacheLRU(object):
    def test_get_session_user(self, postgresql):

        cur = postgresql.cursor()
        cur.execute(
            'CREATE TABLE test (id serial PRIMARY KEY, num integer, data varchar);')
        postgresql.commit()
        cur.close()

        assert lru.get_session_user(None) is None
```

**Comment by fizyk** [comment]
Hi @dmitry-saritasa 

Are your test also running inside docker or outside? We haven't been testing these fixtures with postgresql inside a container.

What settings are you using for starting postrgesql in tests?

**Comment by dmitry-saritasa** [comment]
@fizyk  - the tests are running locally, postgresql is in docker container with port bindings 5432:5432, so any local cli (like psql, pg_dump) work just fine.

**Comment by fizyk** [comment]
I haven't been using docker that much, so I might be getting something wrong:

- You bind the 5432 port to docker, so the port is taken
- You've got a running docker container with running postgresql there on 5432 port. on which it accepts connections?

If that's the case then the problem is that the pytest-postgresql expects to start the postgresql process itself on a free port. Could you share code samples that will help me dubg the issue?

**Comment by douardda** [comment]
This problem might be a locale related one. Since the fixture check the output of pg_xxx command, it depends on the locales, especially the LC_ALL environment variable.

The fixture should take care of setting this environment variable properly. I was looking at this to fill a PR, but I'm not sure how to set the LC_ALL environment variable, since setting it may have side effects on executed unit tests. So it should be set only for the purpose of running pg_ commands.

Reading this code https://github.com/ClearcodeHQ/mirakuru/blob/master/src/mirakuru/base.py#L153 miraku does pass a modified env vars directory to the subprocess call, but does not provide an API to deal with this from Executor's caller side. 

So a proper fix for this problem involve a small refactoring of miraku and a fix in pytest-postgresql, I guess.

**Comment by LienM** [comment]
Hi @fizyk, is there any way to not have pytest-postgresql start the postgresql process, but instead refer to a live one (potentially in a different container?). I'm having the same issue.

**Comment by fizyk** [comment]
You'd need to provide kind of mock fixture that returns objwect with proper attributes mimicking what client fixture needs to connect to it:

See what you'll need: https://github.com/ClearcodeHQ/pytest-postgresql/blob/master/src/pytest_postgresql/factories.py#L274

then in conftest.py have:

```
postgresql = factories.postgresql('mocked_postgresql_proc')

```

That's for now. I think next steps would be to extend client factory with required arguments and possibility to not set the process fixture.

**Comment by LienM** [comment]
Hi @fizyk ,

Thanks! I'll try that and if I can get it to work cleanly maybe I can contribute towards extending the client factory.

**Comment by fizyk** [comment]
@LienM the feature landed on master, and should be part of the 2.2.0 released in few minutes

**Comment by LienM** [comment]
@fizyk Sorry I missed that notification. I'm a little late but thx so much 🙏
